doc/source/rllib/rllib-offline.rst

```diff
@@ -5,437 +5,2081 @@
+        # passed into the `ray.data.Dataset.map_batches` method when sampling
+        # without checking. If no arguments passed in the default arguments
+        # `{'concurrency': max(2, num_learners), 'zero_copy_batch': True}` is
+        # used. Use these keyword args together with `input_read_method_kwargs`
+        # and `iter_batches_kwargs` to tune the performance of the data pipeline.
+        map_batches_kwargs: Optional[Dict],
+        # Keyword args for the `iter_batches` method. These are
+        # passed into the `ray.data.Dataset.iter_batches` method when sampling
+        # without checking. If no arguments are passed in, the default argument
+        # `{'prefetch_batches': 2}` is used. Use these keyword args
+        # together with `input_read_method_kwargs` and `map_batches_kwargs` to
+        # tune the performance of the data pipeline.
+        iter_batches_kwargs: Optional[Dict],
+        # An optional `OfflinePreLearner` class that's used to
+        # transform data batches in `ray.data.map_batches` used in the
+        # `OfflineData` class to transform data from columns to batches that can
+        # be used in the `Learner.update...()` methods. Override the
+        # `OfflinePreLearner` class and pass your derived class in here, if you
+        # need to make some further transformations specific for your data or
+        # loss. The default is `None`` which uses the base `OfflinePreLearner`
+        # defined in `ray.rllib.offline.offline_prelearner`. 
+        prelearner_class: Optional[Type],
+        # An optional `EpisodeReplayBuffer` class is
+        # used to buffer experiences when data is in `EpisodeType` or
+        # RLlib's previous `SampleBatch` type format. In this case, a single
+        # data row may contain multiple timesteps and the buffer serves two
+        # purposes: (a) to store intermediate data in memory, and (b) to ensure
+        # that exactly `train_batch_size_per_learner` experiences are sampled
+        # per batch. The default is RLlib's `EpisodeReplayBuffer`.
+        prelearner_buffer_class: Optional[Type],
+        # Optional keyword arguments for intializing the
+        # `EpisodeReplayBuffer`. In most cases this is simply the `capacity`
+        # for the default buffer used (`EpisodeReplayBuffer`), but it may
+        # differ if the `prelearner_buffer_class` uses a custom buffer.
+        prelearner_buffer_kwargs: Optional[Dict],
+        # Number of updates to run in each learner
+        # during a single training iteration. If None, each learner runs a
+        # complete epoch over its data block (the dataset is partitioned into
+        # at least as many blocks as there are learners). The default is `None`.
+        # This must be set to `1`, if a single (local) learner is used.
+        dataset_num_iters_per_learner: Optional[int],
+    )
 
 Output API
 ----------
 
 You can configure experience output for an agent using the following options:
```

Comment:
nit: let's write 1-2 more sentences here on what exactly gets configured in this method and why?
* Use this method for configuring how any algorithm in RLlib (online- or offline) writes its collected experiences to a data storage.